### PR TITLE
Fix #17878 (COFF fuzz segfault)

### LINKED ISSUE #17878: COFF segfault on test/bins/fuzzed/r2_segfault4_r_coff_symbol_name
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Ubuntu x86 64
| File format of the file you reverse (mandatory)      | COFF
| Architecture/bits of the file (mandatory)            | -
| r2 -v full output, **not truncated** (mandatory)     | https://github.com/radareorg/radare2/commit/8d0711dcc97c3d43a38f02777e192db5269f69bc
### Actual behavior

```
[XX] /home/runner/work/radare2/radare2/test/bins/fuzzed <fuzz> /home/runner/work/radare2/radare2/test/bins/fuzzed/r2_segfault4_r_coff_symbol_name
make: *** [fuzz-tests] Error 1
R2_NOPLUGINS=1 radare2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc aaa /home/runner/work/radare2/radare2/test/bins/fuzzed/r2_segfault4_r_coff_symbol_name
-- stdout

-- stderr

-- exit status: -1
[**] /home/runner/work/radare2/radare2/test/bins/fuzzed      504 OK         0 BR        1 XX        0 FX

Finished in 5 minutes and 15 seconds.
Makefile:15: recipe for target 'fuzz-tests' failed
Error: Process completed with exit code 2.
```
And with ASAN:
```
[XX] /home/runner/work/radare2/radare2/test/bins/fuzzed <fuzz> /home/runner/work/radare2/radare2/test/bins/fuzzed/r2_segfault4_r_coff_symbol_name
R2_NOPLUGINS=1 radare2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc aaa /home/runner/work/radare2/radare2/test/bins/fuzzed/r2_segfault4_r_coff_symbol_name
-- stdout

-- stderr
../libr/bin/p/bin_coff.c:195:28: runtime error: load of null pointer of type 'long long unsigned int'
AddressSanitizer:DEADLYSIGNAL
=================================================================
==28223==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x7f80b2b44f66 bp 0x7ffe96326aa0 sp 0x7ffe96326a50 T0)
==28223==The signal is caused by a READ memory access.
==28223==Hint: address points to the zero page.
    #0 0x7f80b2b44f65 in sections ../libr/bin/p/bin_coff.c:195
    #1 0x7f80b2b13154 in r_bin_object_set_items ../libr/bin/bobj.c:345
    #2 0x7f80b2b0ea67 in r_bin_object_new ../libr/bin/bobj.c:174
    #3 0x7f80b2b03460 in r_bin_file_new_from_buffer ../libr/bin/bfile.c:524
    #4 0x7f80b2abff6a in r_bin_open_buf ../libr/bin/bin.c:286
    #5 0x7f80b2ac0d9f in r_bin_open_io ../libr/bin/bin.c:346
    #6 0x7f80b589de04 in r_core_file_do_load_for_io_plugin ../libr/core/cfile.c:436
    #7 0x7f80b58a0840 in r_core_bin_load ../libr/core/cfile.c:651
    #8 0x7f80bd6c082a in r_main_radare2 ../libr/main/radare2.c:1106
    #9 0x560ff835703b in main ../binr/radare2/radare2.c:96
    #10 0x7f80bca600b2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x270b2)
    #11 0x560ff8356acd in _start (/home/runner/bin/radare2+0x8aacd)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV ../libr/bin/p/bin_coff.c:195 in sections
==28223==ABORTING
```

cc @pelijah since you worked on COFF recently.

### PR DESCRIPTION
**Pull request by pelijah**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: libr/bin/format/coff/coff.c [hunk 1]
```diff
@@ -224,12 +224,12 @@ static int r_bin_coff_init(struct r_bin_coff_obj *obj, RBuffer *buf, bool verbos
 		bprintf ("Warning: failed to init section header\n");
 		return false;
 	}
-	if (!r_bin_coff_init_symtable (obj)) {
-		bprintf ("Warning: failed to init symtable\n");
+	if (!r_bin_coff_init_scn_va(obj)) {
```
Comment (trufae) [inline libr/bin/format/coff/coff.c:227]: missing space before (

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/bin/format/coff/coff.c (hunk 1 above)
Comment (trufae) [inline libr/bin/format/coff/coff.c:227]:
```suggestion
	if (!r_bin_coff_init_scn_va (obj)) {
```

**Review by trufae — CHANGES_REQUESTED** [review]
.

**Comment by trufae** [comment]
Make the pr title more descriptive

**Comment by kazarmy** [comment]
Please put `fuzz` in the branch name so that the fuzz tests are run. Even better, put `asanfuzz` so that asan is run on the fuzz tests.

**Review by kazarmy — APPROVED** [review]
Not sure whether this works but let's give it a try!

**Pull request merged**